src/yunohost/app.py

```diff
@@ -89,27 +90,24 @@ def app_fetchlist(url=None, name=None):
 
     """
     # Create app path if not exists
-    try: os.listdir(repo_path)
-    except OSError: os.makedirs(repo_path)
+    if not os.path.exists(repo_path):
+        os.makedirs(repo_path)
 
     if url is None:
         url = 'https://app.yunohost.org/official.json'
         name = 'yunohost'
-    else:
-        if name is None:
-            raise MoulinetteError(errno.EINVAL,
-                                  m18n.n('custom_appslist_name_required'))
+    elif name is None:
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('custom_appslist_name_required'))
 
-    list_file = '%s/%s.json' % (repo_path, name)
-    if os.system('wget "%s" -O "%s.tmp"' % (url, list_file)) != 0:
-        os.remove('%s.tmp' % list_file)
+    try:
+        urlretrieve(url, '%s/%s.json' % (repo_path, name))
+    except Exception as e:
+        # I don't know how to put e into the MoulinetteError stuff
+        print e
```

Comment:
Just use the `logger` object instead to log the exception traceback - and error message too, i.e. `logger.debug('cannot fetch app list', exc_info=1)`.
Or, if you want to display an English error message to the user - who will probably not understand what it is about - you can update the `appslist_retrieve_error` key [here](https://github.com/YunoHost/yunohost/blob/unstable/locales/en.json#L14), append for example `: {error}` and pass this argument value with `m18n.n('appslist_retrieve_error', error=str(e))`.